damnit/ctxsupport/damnit_ctx.py

```diff
@@ -69,3 +79,67 @@ def annotations(self):
         Returns a dict of argument names to their annotations.
         """
         return getattr(self.func, '__annotations__', {})
+
+
+class Cell:
+    """Variable functions can return this"""
+    def __init__(self, data, summary=None, summary_value=None, bold=None, background=None):
+        # If the user returns an Axes, save the whole Figure
+        if isinstance(data, Axes):
+            data = data.get_figure()
+
+        if not isinstance(data, (xr.Dataset, xr.DataArray, str, type(None), Figure)):
+            data = np.asarray(data)
+            # Numpy will wrap any Python object, but only native arrays
+            # can be saved in HDF5, not those containing Python objects.
+            if data.dtype.hasobject:
+                raise TypeError(f"Returned data type {type(data)} cannot be saved")
+
+        self.data = data
+        self.summary = summary
+        self.summary_value = summary_value
+        self.bold = bold
+        self.background = self._normalize_colour(background)
+
+    @staticmethod
+    def _normalize_colour(c):
+        if isinstance(c, str):
+            if not re.match(r'#[0-9A-Fa-f]{6}', c):
+                raise ValueError("Colour string should be hex code (like '#ffcc00')")
+            b = bytes.fromhex(c[1:])
+            return np.frombuffer(b, dtype=np.uint8)
+        elif isinstance(c, Sequence):
+            if not len(c) == 3:
+                raise TypeError(f"Wrong number of values ({len(c)}) for R,G,B")
+            if not all(0 <= v <= 255 for v in c):
+                raise ValueError("Colour values must be 0 - 255")
+            return np.array(c, dtype=np.uint8)
```

Comment:
Is there a benefit to storing it as a uint8 array? I know that's an appropriate type, but I don't think there's a technical reason in this case, and if we stored it as a `list` instead we wouldn't need to handle ndarray's specially in `get_attrs()`.

Reply:
Between this code (in the context process) and `get_attrs` it's stored as an HDF5 attribute, which h5py will always read as a numpy array.